src/components/elements/DidSearch.tsx

```diff
@@ -73,8 +73,8 @@ export const DidSearch: React.FunctionComponent<DidSearchProps> = (props) => {
       <Search color={colors.primary.base} size="24" style={{ position: "absolute", left: "10px", top: "12px" }} />
       <Input
         borderRadius={isOpen ? "4px 4px 0 0" : 1}
-        onChange={(event: any) => {
-          onChange(event.target.value);
+        onChange={(event: React.ChangeEvent<HTMLInputElement>) => {
+          onChange({ did: event.target.value });
```

Comment:
I usually don't think to add types to `event`, but in this case it would have caught the error, because it would know that `event.target.value` is a `string`, and that this `onChange` function is typed to only accept `SelectedDid` type, not `string`s.